ublox_serialization/include/ublox_serialization/serialization.hpp

```diff
@@ -532,9 +538,13 @@ class Reader {
     return (classId() == class_id && messageId() == message_id);
   }
 
- private:
+  const std::string &getUnusedData() const { return unused_data_; }
+
+private:
   //! The buffer of message bytes
   const uint8_t *data_;
+  //! Unused data from the read buffer, contains nmea messages.
+  std::string unused_data_;
```

Comment:
This isn't really unused data, right?  It is more "extra" data that contains some additional metadata.  Maybe we should name it `extra_data` instead?

Reply:
Agreed, "unused" doesn't feel quite appropriate, since it was used for both NMEA strings and, presumably, any ubx messages without a subscription.

I'm fine with `extra_data`, or perhaps `unmatched_data` or `unprocessed_data`.

Reply:
Yeah, I'd say let's go with `extra_data`, and update everything else accordingly.